theano/gpuarray/opt.py

```diff
@@ -252,12 +253,25 @@ def local_opt(node):
                 # This is needed as sometimes new_op inherits from OP.
                 if new_op and new_op != node.op:
                     if isinstance(new_op, theano.Op):
```

Comment:
Should the call to maker on line 251 should be in a "with inherit_stack_trace", as sometimes it will build intermediate nodes. For example see the local_gpua_careduce optimization

I don't know if we should do that only when there is only 1 outputs (or all outputs have the same stack trace) or we should always do that and use the first stack trace? Currently this PR modify the local_gpua_careduce opt to manually add the stack trace. So this is more for a follow up PR.